saleor/graphql/checkout/mutations/checkout_add_promo_code.py

```diff
@@ -74,7 +74,10 @@ def perform_mutation(  # type: ignore[override]
         manager = get_plugin_manager_promise(info.context).get()
         lines, unavailable_variant_pks = fetch_checkout_lines(checkout)
 
-        if unavailable_variant_pks:
+        if (
+            unavailable_variant_pks
+            and checkout.channel.use_legacy_error_flow_for_checkout
```

Comment:
In this place, we execute additional DB query for `channel`. Channel is also fetched in this flow in line 96.
`checkout_info = fetch_checkout_info(`

Reply:
DB queries shows the same amount of queries ☝️ 
I also double-check it locally by making a db benchmark vs main.